service/src/main/java/bio/terra/landingzone/stairway/flight/ResourceNameProvider.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.landingzone.stairway.flight;
+
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.stairway.flight.create.resource.step.BaseResourceCreateStep;
+import bio.terra.landingzone.stairway.flight.exception.ResourceNameGenerationException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/** Provides name for a resource during LZ resource creation flight. */
```

Comment:
I think it'd be worth expressing the constraints this class is placing on the universe of resource type names. Specifically that only 1 instance of each resource type can get a name and is constrained by the length expressed in a `ResourceNameRequirements`. I _think_ that should be ok for now -- if we ever need more than one of something we'll have to rethink.

Reply:
Extend comments.